src/encoded/types/file.py

```diff
@@ -1459,98 +1459,74 @@ def update_google_analytics(context, request, ga_config, filename, file_size_dow
                                                        ga_config["hostnameTrackerIDMapping"].get("default"))
+    ga_tid = ga_tid_mapping[1] if isinstance(ga_tid_mapping, list) and len(ga_tid_mapping) > 1 else None
+
     if ga_tid is None:
         raise Exception("No valid tracker id found in ga_config.json > hostnameTrackerIDMapping")
 
-    # We're sending 2 things here, an Event and a Transaction of a Product. (Reason 1 for redundancies)
-    # Some fields/names are re-used for multiple things,
-    # such as filename for event label + item name dimension + product name + page title dimension (unusued) + ...
+    file_extension = file_extension = os.path.splitext(filename)[1][1:]
+
     ga_payload = {
-        "v": 1,
-        "tid": ga_tid,
-        "t": "event",  # Hit type. Could also be event, transaction, pageview, etc.
-        # Override IP address. Else will send detail about EC2 server which not too useful.
-        "uip": request.remote_addr,
-        "ua": request.user_agent,
-        "dl": request.url,
-        "dt": filename,
-
-        # This cid below is a ~ random ID/number (?). Used as fallback, since one is required
-        # if don't provided uid. While we still allow users to not be logged in,
-        # should at least be able to preserve/track their anon downloads..
-
-        # '555' is in examples and seemed to be referred to as example for anonymous sessions in some Google doc.
-        # But not 100% sure and wasn't explicitly stated to be "555 for anonymous sessions" aside from usage in example.
-        # Unsure if groups under 1 session or not.
-        "cid": "555",
-        "an": "4DN Data Portal EC2 Server",  # App name, unsure if used yet
-        "ec": "Serverside File Download",  # Event Category
-        "ea": "Range Query" if request.range else "File Download",  # Event Action
-        "el": filename,  # Event Label
-        "ev": file_size_downloaded,  # Event Value
-        # Product fields
-        "pa": "purchase",
-        "ti": str(uuid4()),  # We need to send a unique transaction id along w. 'transactions' like purchases
-        "pr1id": file_at_id,  # Product ID/SKU
-        "pr1nm": filename,  # Product Name
-        "pr1br": lab.get("display_title"),  # Product Branch
-        "pr1qt": 1,  # Product Quantity
-        # Product Category from @type, e.g. "File/FileProcessed"
-        "pr1ca": "/".join([ty for ty in reversed(context.jsonld_type()[:-1])]),
-        # Product "Variant" (supposed to be like black, gray, etc), we repurpose for filetype for reporting
-        "pr1va": file_type
-        # "other" MATCHES THAT IN `file_type_detaild` calc property,
-        # since file_type_detailed is used on frontend when performing "Select All" files.
+        "client_id": ga_cid,
+        "timestamp_micros": str(int(datetime.datetime.now().timestamp() * 1000000)),
+        "non_personalized_ads": False,
+        # "user_agent": request.user_agent,
```

Comment:
Is this not necessary?

Reply:
`client_id `is required to identify the user's session, whereas `timestamp_micros ` determines when the event is triggered.